routing/turns_tts_text.cpp

```diff
@@ -62,20 +297,150 @@ std::string GetTtsText::GetTurnNotification(Notification const & notification) c
+      if (!dirExitStr.empty()) {
+        dirStr = std::move(dirExitStr);
+        ontoStr = ""; // take_exit_number overwrites "onto"
+      }
+    }
+
+    // Same as above but for dirStr instead of distStr
+    strings::ReplaceLast(dirStr, ".", "");
+    strings::ReplaceLast(dirStr, "。", "");
+    strings::ReplaceLast(dirStr, "।", "");
+
+    std::string distDirOntoStreetStr = GetTextById("dist_direction_onto_street");
+    // TODO: we may want to only load _street_verb if _street exists; may also need to handle
+    //   a lack of a $5 position in the formatter string
+    std::string dirVerb = GetTextById(dirKey+"_street_verb");
+
+    if (localeKey == "hu") {
+      HungarianBaseWordTransform(streetOut); // modify streetOut's last letter if it's a vowel
+
+      // adjust the -re suffix in the formatter string based on last-word vowels
+      uint8_t hungarianism = CategorizeHungarianLastWordVowels(streetOut);
+      if (hungarianism == 1) {
+        strings::ReplaceLast(distDirOntoStreetStr, "-re", "re"); // just remove hyphenation
+      } else if (hungarianism == 2) {
+        strings::ReplaceLast(distDirOntoStreetStr, "-re", "ra"); // change re to ra without hyphen
+      } else {
+        strings::ReplaceLast(distDirOntoStreetStr, "-re", ""); // clear it
+      }
+
+      // if the first pronounceable character of the street is a vowel, use "az" instead of "a"
+      // 1, 5, and 1000 start with vowels but not 10 or 100 (including 5*, 5**, 1*, 1**, 1***, etc)
+      static const std::regex rHun("^[5aeiouyáéíóúöüőű]|^1$|^1[^\\d]|^1\\d\\d\\d[^\\d]", std::regex_constants::icase);
+      std::smatch ma;
+      if (std::regex_search(streetOut, ma, rHun) && ma.size() > 0) {
+        if (ontoStr == "a")
+          ontoStr = "az";
+        if (dirStr == "Hajtson ki a")
+          dirStr = "Hajtson ki az";
+      }
+    }
+
+    char ttsOut[1024];
+    snprintf(ttsOut, sizeof(ttsOut)/sizeof(ttsOut[0]),
+      distDirOntoStreetStr.c_str(),
+      distStr.c_str(), // in 100 feet
+      dirStr.c_str(), // turn right / take exit
+      ontoStr.c_str(), // onto / null
+      streetOut.c_str(), // Main Street / 543:: M4: Queens Parkway, London
+      dirVerb.c_str() // (optional "turn right" verb)
+    );
+
+    // remove floating punctuation
+    static const std::regex rP(" [,\\.:;]+ ");
+    std::string cleanOut = std::regex_replace(ttsOut, rP, " ");
+    // remove repetitious spaces or colons
+    static const std::regex rS("[ :]{2,99}");
+    cleanOut = std::regex_replace(cleanOut, rS, " ");
+    // trim leading spaces
+    static const std::regex rL("^ +");
+    cleanOut = std::regex_replace(cleanOut, rL, "");
+
+    LOG(LINFO, ("TTSn", thenStr + cleanOut));
+
+    return thenStr + cleanOut;
+  }
+
+  std::string out;
+  if (!distStr.empty()) {
+    // add distance and/or space only if needed, for appropriate languages
+    if(localeKey != "ja")
+      out = thenStr + distStr + " " + dirStr;
```

Comment:
Do spaces break ja TTS?

Reply:
It's not about breaking, simply that Japanese doesn't have spaces and stray spaces make the test suite weird/difficult/inconsistent.